distributed/cli/dask_scheduler.py

```diff
@@ -70,11 +72,14 @@ def del_pid_file():
         with ignoring(ImportError):
             from distributed.bokeh.scheduler import BokehScheduler
             services[('bokeh', bokeh_internal_port)] = BokehScheduler
-    scheduler = Scheduler(loop=loop, services=services)
+    scheduler = Scheduler(loop=loop, services=services,
+                          scheduler_file=scheduler_file)
     scheduler.start(addr)
 
     bokeh_proc = None
     if _bokeh:
+        if bokeh_port == 0:
+            bokeh_port = open_port()
```

Comment:
The problem is that bokeh is running in a separate process, and we would like to communicate the address back to the main process.  This choice was made as an easy hack that solved the existing problem most of the time.  A better solution would be ideal though.